packages/engine-server/src/drivers/sqlite/SqliteMetadataStore.ts

```diff
@@ -30,6 +39,94 @@ import { SchemaNotesTableUtils } from "./tables/SchemaNotesTableUtils";
 export class SqliteMetadataStore implements IDataStore<string, NotePropsMeta> {
   constructor(private _db: Database, private _vaults: DVault[]) {}
 
+  /**
+   * Goes through all domains and recursively apply schemas.
+   */
+  async initSchema(fileStore: IFileStore, wsRoot: string, logger: DLogger) {
+    // this whole thing to parseAllNoteFilesForSqlite / SqliteDbFactory
+    const schemaResult = await SqliteDbFactory.initSchema(
+      this._vaults,
+      wsRoot,
+      fileStore,
+      logger
+    );
+    if (schemaResult.isErr()) {
+      throw schemaResult.error;
+    }
+    const schemas = schemaResult.value;
+    const rootNotesResult = await this.query({ qs: "", originalQS: "" });
+    if (rootNotesResult.isErr()) {
+      throw rootNotesResult.error;
+    }
+    const rootNotes = rootNotesResult.value;
+    const rootNoteIds = rootNotes.map((rootNoteMeta) => {
+      return rootNoteMeta.id;
+    });
+
+    // TODO: break this down
+    await Promise.all(
+      rootNoteIds.map(async (rootId) => {
+        const rootNoteResp = await this.get(rootId);
+        if (rootNoteResp.error) {
+          throw rootNoteResp.error;
+        } else {
+          const domainIds = rootNoteResp.data.children;
+          const allNoteIdsInDomains = _.flatten(
+            await Promise.all(
+              domainIds.map(async (domainId) => {
+                const getAllInDomainResult =
+                  await HierarchyTableUtils.getAllInDomain(this._db, domainId);
+                if (getAllInDomainResult.isErr()) {
+                  throw getAllInDomainResult.error;
+                }
+                const allInDomainRows = getAllInDomainResult.value;
+                return [domainId].concat(
+                  allInDomainRows.map((row) => {
+                    return row.child;
+                  })
+                );
+              })
+            )
+          );
+          // this is too expensive
```

Comment:
How long does this schema setup take for org-private? (maybe we do this measurement/tuning as a follow up)  If it helps, I left a test-case snippet that can be locally run to get an idea of how long things take here: [[user.jonathan.dev.sqlite-temp-code]]

Might need to add some additional logging though (I also am working on a pr to improve some of our logging in parseAllNoteFiles()

Reply:
have to test that out. I'll let you know.

Reply:
Seems like `SqliteMetadataStore.initSchema` take about 2500ms consistenly on my machine with `org-workspace`.

For comparison on my machine it take about 8 seconds to for delta init + schema init to finish. so about 30% of the time is spent on `initSchema`.